_posts/2018-03-28-building-coreboot-on-minnowboard.md

```diff
@@ -0,0 +1,244 @@
+Building coreboot on MinnowBoard
+================================
+
+Building coreboot on well supported platform such as Bay Trail is quite
+straightforward task, however we need to remember about some things in
+order to have coreboot working. First of all we need to provide up-to-date
+microcode and FSP (Firmware Support Package), which are not included in
+coreboot source tree and coreboot build system won't complain about it.
+Second thing is that Bay Trail has ME firmware on the same ROM as boot firmware
+so we have to make sure that we won't corrupt it because it would brick the
+platform. Except those we have standard procedure, we need to get a toolchain.
+
+Microcode
+---------
+
+Newest microcode can be found on https://firmware.intel.com/projects/minnowboard-max
+We have to provide it because it's a condition for warranty for CPU.
+It is provided as Intel-syntax assembly file with microcode as though
+it was ordinary data:
+
+```asm
+dd 000000001h ; Header Version
+dd 000000321h ; Patch ID
+dd 001142014h ; DATE
+dd 000030673h ; CPUID
+dd 0984d6012h ; Checksum
+dd 000000001h ; Loader Version
+dd 000000001h ; Platform ID
+dd 00000cbd0h ; Data size
+dd 00000cc00h ; Total size
+dd 000000000h ; reserved
+dd 000000000h ; reserved
+dd 000000000h ; reserved
+dd 000000000h
+dd 0000000a1h
+dd 000020001h
+dd 000000321h
+```
+
+details of it's structure are confidential. However for coreboot we need to
+provide it as comma separated list of values in C-style format, like this:
+
+```C
+0x000000001,
+0x00000021d,
+0x008122013,
+0x000030672,
+0x0a565e78c,
+0x000000001,
+0x000000002,
+0x00000cbd0,
+0x00000cc00,
+0x000000000,
+0x000000000,
+0x000000000,
+0x000000000,
+0x0000000a1,
+```
+
+We can easly download and convert it using simple script:
+
+```bash
+wget 'http://intel.ly/2yStb2e' -O MM-firmware.zip
+unzip MM-firmware.zip
+cd MinnowBoard_MAX-0.97-Binary.Objects/Vlv2MiscBinariesPkg/Microcode
+for f in *.inc ; do
+    name=${f%.inc}
+    awk '{gsub( /h.*$/, "", $2 ); print "0x" $2 ","; }' $f \
+        > $name.h
+done
+cp *.h ../../../
+```
+
+FSP (Firmware Support Package)
+---
+
+FSP is binary package from Intel dedicated to firmware developers containing
+most important platform initialization (including IP). However there is non-FSP
+version of coreboot for Bay Trail however it won't work without MCU code which
```

Comment:
@BartekLew MCU code?

Reply:
@pietrushnic you told there is something that enables build without the FSP. I propably mistook the name?

Reply:
@BartekLew you mean MRC code (Memory Reference Code), this is what I saw at some point but it would be great if you can confirm that in coreboot source tree. I believe there is version that can be build with mrc.bin or sth. Please check `HAVE_MRC` in `src/soc/intel/baytrail/Kconfig`

Reply:
```
69 config HAVE_MRC
 70     bool "Add a Memory Reference Code binary"
 71     default y
 72     help
 73       Select this option to add a blob containing
 74       memory reference code.
 75       Note: Without this binary coreboot will not work
```
Here it is.